src/coreclr/jit/morph.cpp

```diff
@@ -12044,7 +12044,7 @@ GenTree* Compiler::fgMorphFieldToSimdGetElement(GenTree* tree)
     unsigned    simdSize        = 0;
     GenTree*    simdStructNode  = getSIMDStructFromField(tree, &simdBaseJitType, &index, &simdSize);
 
-    if (simdStructNode != nullptr)
+    if ((simdStructNode != nullptr) && IsBaselineSimdIsaSupported())
```

Comment:
I wonder if should call `fgMorphFieldToSimdGetElement` at all when `!IsBaselineSimdIsaSupported()`

Reply:
I can move the check further up.

Reply:
This was moved out for both `fgMorphFieldToSimdGetElement` and `fgMorphFieldAssignToSimdSetElement` so we only do the transforms when the baseline ISAs are marked supported.